src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,99 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Encapsulates the list of tasks that the user is currently tracking. Analogous to an ArrayList and supports
+ * the relevant operations.
+ */
+public class TaskList {
+
+    private final ArrayList<Task> list = new ArrayList<>();
```

Comment:
Perhaps would be better if the variable name here be `lists`?

Reply:
Noted on the naming violation. Will rename to `tasks` in the next commit.